UniversalAlgebra/GaloisConnections/Basic.lagda

```diff
@@ -39,46 +42,57 @@ module _ {α α₁ α₂ : Level}(A : Poset α α₁ α₂)
   _≤A_ = _≤_ A
   _≤B_ = _≤_ B
 
- record Galois : Type  (α ⊔ α₂ ⊔ β ⊔ β₂) where
+ record Galois : Type (lsuc ℓ)  where
   field
    F : Carrier A → Carrier B
    G : Carrier B → Carrier A
-   F⊣G : ∀ a b → (F a) ≤B b → a ≤A (G b)
-   F⊢G : ∀ a b → a ≤A (G b) → (F a) ≤B b
+   GF≥id : ∀ a →  a ≤A G (F a)
+   FG≥id : ∀ b →  b ≤B F (G b)
+
+ record Residuation : Type (lsuc ℓ)  where
+  field
+   f : Carrier A → Carrier B
+   g : Carrier B → Carrier A
+   gf≥id : ∀ a → a ≤A g (f a)
+   fg≤id : ∀ b → f (g b) ≤B b
 
 
-private variable ℓ ρ : Level
 
+module _ {ℓ : Level}{𝒜 ℬ : Type ℓ} where
 
-module _ {𝒜 ℬ : Type ℓ} where
+ infix 10 _⃗_ _⃖_
```

Comment:
While I totally love the idea of left and right residuals, is this the right concept? [I'd be happy if the answer is an explainable 'yes'!]

Reply:
I'm not quite sure what you mean here.  I added a definition of a Residuation record type to represent a pair of residuated mappings for fun because these are related to, but not the same as, Galois connections.  (If you reverse the order of one of the partial sets involved---i.e., consider the opposite category---then you can convert a Residual pair into a Galois pair.) So, in that sense you're right, residuation is not the right concept to use when defining a Galois connection.

However, I'm confused by the lines of code you cite with your comment.  The operations  _⃗_ _⃖_ are not, as far as I can tell, left and right residuals.  They are the correct maps to use when relating a binary relation to the Galois correspondence that it induces (as I prove in the GaloisConnections/Properties module).  Perhaps there's not enough here yet to justify two separate modules... but I like your style (and that of the std lib) of generally keeping modules fairly short and focused.

By the way, I forgot one condition for the Residuation type.  At least one of the maps (whence both) must be order preserving (a morphism in the category of posets).  I'll add that to the next PR.

Reply:
I got "fooled" by the notation.  My error.

On the Zulip, people thought the two operations seemed like a terminal/initial condition (someone else brought up pushout/pullback). Just the operation, without the universality condition, of course.